GrandCentralBoard/Widgets/Bonus/Extensions/Array.swift

```diff
@@ -18,3 +18,17 @@ extension RangeReplaceableCollectionType where Generator.Element : Equatable {
     }
 
 }
+
+extension SequenceType where Generator.Element == Bonus {
+    func sortByDate() -> [Bonus] {
+        return self.sort({ $0.date.compare($1.date) == .OrderedDescending })
+    }
+
+    func flatten() -> [Bonus] {
+        let mappedBonuses = self.map { bonus in
+            return [ Bonus(name: bonus.name, amount: bonus.amount, receiver: bonus.receiver, date: bonus.date, childBonuses: [])]
```

Comment:
Do we really need to produce fantom `Bonus` object by losing `childBonuses` information? What's wrong with `return [bonus] + bouns.childBonuses`?

Also, `map({ transform }).flatMap({ $0 })` can be simplified to `flatMap({ transform })`.